### PR TITLE
[ENH] API change: nilearn.stats -> nilearn.glm

### PR DESCRIPTION
**Pull request by tbng**
Closes gh-2472

Changes proposed in this pull request:
* API change: `nilearn.stats -> nilearn.glm`
* API change (cont.): `nilearn.stats.first_level_model` -> `nilearn.glm.first_level`
* API change (cont.): `nilearn.stats.second_level_model` -> `nilearn.glm.second_level`
* Various indentation fixes for `nilearn.glm` example

Edit:
Also closes gh-2490 with incorporating the following changes:
* API change: `nilearn.stats.map_threshold` to `nilearn.glm.threshold_stats_img` (closed PR gh-2498 for trying to do the same change).

### CODE REVIEW

File: examples/04_glm_first_level/plot_adhd_dmn.py [hunk 1]
```diff
@@ -18,8 +17,8 @@
 from nilearn import datasets, plotting
 from nilearn.input_data import NiftiSpheresMasker
 
-from nilearn.stats.first_level_model import FirstLevelModel
-from nilearn.stats.first_level_model import make_first_level_design_matrix
+from nilearn.glm.first_level import FirstLevelModel
```
Comment (emdupre) [inline examples/04_glm_first_level/plot_adhd_dmn.py:20]: Should we also rename `FirstLevelModel` to `FirstLevel` to match ?

**Comment by tbng** [comment]
I had a thought about that but I for class name I think sticking with
`FirstLevelModel` and `SecondLevelModel` is still more clear, as reducing
`nilearn.glm.first_level_model` to nilearn.glm.first_level should be concise
enough?

File: examples/04_glm_first_level/plot_adhd_dmn.py (hunk 1 above)
Comment (effigies) [reply examples/04_glm_first_level/plot_adhd_dmn.py:20]: I feel like it loses clarity to just call it a `FirstLevel`. If we take [sklearn.linear_model](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.linear_model) as an example, most of them have a second word like "Regression" unless the first part of the name is really iconic like "Lasso". So I'd stick with `FirstLevelModel`.

`FirstLevelGLM` would also be clear.

File: examples/plot_single_subject_single_run.py [hunk 2]
```diff
@@ -21,14 +20,17 @@
 under the direction of Karl Friston. It is provided by FIL methods
 group which develops the SPM software.
 
-According to SPM documentation, 96 scans were acquired (repetition time TR=7s) in one session. The paradigm consisted of alternating periods of stimulation and rest, lasting 42s each (that is, for 6 scans). The sesssion started with a rest block.
-Auditory stimulation consisted of bi-syllabic words presented binaurally at a
-rate of 60 per minute. The functional data starts at scan number 4, that is the
-image file ``fM00223_004``.
+According to SPM documentation, 96 scans were acquired (repetition time TR=7s)
+in one session. The paradigm consisted of alternating periods of stimulation
+and rest, lasting 42s each (that is, for 6 scans). The sesssion started with a
```
Comment (bthirion) [inline examples/plot_single_subject_single_run.py:25]:
```suggestion
and rest, lasting 42s each (that is, for 6 scans). The session started with a
```

**Comment by bthirion** [comment]
Fine with keeping `FirstLevelModel` and `SecondLevelModel` .

**Comment by GaelVaroquaux** [comment]
> Fine with keeping FirstLevelModel and SecondLevelModel .

Me too. I agree that it is more clear.

@bthirion : looking a bit more at this, I think that we should remove one
level of nesting and expose all objects in nilearn.glm. There are not too
many classes and functions defined in the submodules, and it will make it
easier for the user, following the well-established design principle that
"flat is better than nested".

Would you be OK with such change?

**Comment by bthirion** [comment]
> Would you be OK with such change?

Yes, I had that in mind, because the submodule is not so populated.
Let's do that in the next PR

File: examples/04_glm_first_level/plot_fir_model.py [hunk 3]
```diff
@@ -5,7 +5,11 @@
 The example below shows that they're good to do statistical inference
 even on fast event-related fMRI datasets.
 
-Here, we demonstrate the use of a FIR model with 3 lags, computing 4 contrasts from a single subject dataset from the "Neurospin Localizer". It is a fast event related design: During 5 minutes, 80 events of the following types are presented : ['calculaudio', 'calculvideo'', 'clicDvideo', 'clicGaudio', 'clicGvideo', 'damier_H', 'damier_V', 'phraseaudio', 'phrasevideo']
+Here, we demonstrate the use of a FIR model with 3 lags, computing 4 contrasts
+from a single subject dataset from the "Neurospin Localizer". It is a fast
+event related design: During 5 minutes, 80 events of the following types are
+presented : ['calculaudio', 'calculvideo'', 'clicDvideo', 'clicGaudio',
+'clicGvideo', 'damier_H', 'damier_V', 'phraseaudio', 'phrasevideo']
```
[the remark this answers is not in the document]
Comment (GaelVaroquaux) [reply examples/04_glm_first_level/plot_fir_model.py:12]: Should we address this here, or in a follow up PR?

File: examples/04_glm_first_level/plot_first_level_details.py [hunk 4]
```diff
@@ -275,14 +278,13 @@ def plot_contrast(first_level_model):
...
 #
-# There seems to be something here. Maybe we could adjust the
-# We could try to go one step further: using not only the so-called
-# canonical hrf, but also its time derivative. Note that in that case,
-# we still perform the contrasts and obtain statistical significance
-# for the main effect ---not the time derivative. This means that the
-# inclusion of a time derivative in the design matrix has the sole
-# effect of discounting timing misspecification from the error term,
-# which would decrease the estimated variance and enhance the
+# There seems to be something here. Maybe we could adjust the We could try to
```
Comment (GaelVaroquaux) [inline examples/04_glm_first_level/plot_first_level_details.py:281]:
```suggestion
# There seems to be something here. Maybe we could try to
```

File: examples/04_glm_first_level/plot_fir_model.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bthirion) [reply examples/04_glm_first_level/plot_fir_model.py:12]: Since this is small, I suggest to do it here.

File: nilearn/glm/thresholding.py [hunk 5]
```diff
@@ -173,8 +172,9 @@ def cluster_level_inference(stat_img, mask_img=None,
     return proportion_true_discoveries_img
 
 
-def map_threshold(stat_img=None, mask_img=None, alpha=.001, threshold=3.,
-                  height_control='fpr', cluster_threshold=0, two_sided=True):
+def threshold_stats_img(stat_img=None, mask_img=None, alpha=.001, threshold=3.,
+                        height_control='fpr', cluster_threshold=0,
+                        two_sided=True):
```
Comment (GaelVaroquaux) [inline nilearn/glm/thresholding.py:177]: We should add a "see also" section to this function that points to nilearn.image.threshold_img, and vice versa.

**Comment by GaelVaroquaux** [comment]
This is almost ready:
* [ ] two suggestions to merge
* [ ] Update the list of events in  examples/04_glm_first_level/plot_fir_model.py 
* [ ] Add see also sections in nilearn.image.threshold_img and nilearn.glm.threshold_stats_img.

**Comment by GaelVaroquaux** [comment]
@tbng : can we resuscitate this PR? :). It's almost done!

Thanks!!

**Comment by tbng** [comment]
@GaelVaroquaux @bthirion I have added a new commit; could you check again? Thanks.

**Comment by bthirion** [comment]
I'll merge tonight unless I hear otherwise.

**Comment by GaelVaroquaux** [comment]
Merging.

**Pull request merged**